apps/meteor/tests/end-to-end/api/chat.ts

```diff
@@ -2945,6 +2945,283 @@ describe('[Chat]', () => {
 			filterDiscussionsByText(text);
 		});
 	});
+
+	describe('[/chat.syncMessages]', () => {
+		let testChannel: IRoom;
+
+		before(async () => {
+			testChannel = (await createRoom({ type: 'c', name: `channel.test.syncMessages.${Date.now()}` })).body.channel;
+		});
+
+		after(() => deleteRoom({ type: 'c', roomId: testChannel._id }));
+
+		it('should return an error when the required "roomId" parameter is not sent', (done) => {
+			void request
+				.get(api('chat.syncMessages'))
+				.set(credentials)
+				.expect('Content-Type', 'application/json')
+				.expect(400)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', false);
+					expect(res.body.errorType).to.be.equal('invalid-params');
+				})
+				.end(done);
+		});
+
+		it('should return an error when the neither "lastUpdate", "next" or "previous" parameter is sent', (done) => {
+			void request
+				.get(api('chat.syncMessages'))
+				.set(credentials)
+				.query({ roomId: 'invalid-room' })
+				.expect('Content-Type', 'application/json')
+				.expect(400)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', false);
+					expect(res.body.errorType).to.be.equal('error-param-not-provided');
+				})
+				.end(done);
+		});
+
+		it('should return an error when the "lastUpdate" parameter is invalid', (done) => {
+			void request
+				.get(api('chat.syncMessages'))
+				.set(credentials)
+				.query({ roomId: 'invalid-room', lastUpdate: 'invalid-date' })
+				.expect('Content-Type', 'application/json')
+				.expect(400)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', false);
+					expect(res.body.errorType).to.be.equal('error-lastUpdate-param-invalid');
+				})
+				.end(done);
+		});
+
+		it('should return an error when user is not allowed to access the room', (done) => {
```

Comment:
I think this title doesn't match the test case, it seems you are providing an invalid room id instead...